File: OEA/OEA.Library/Modeling/Web/Scripts/commands/BackupViewConfig.js
Ext.define('BackupViewConfig', {
    extend: 'Oea.cmd.Command',
    config: {
        meta: {
            text: "重置",
            tooltip: "重置为原始配置"
        }
    },
    constructor: function () {
        this.svc = 'BackupViewConfigService';
    },
    execute: function (view) {
        var c = view.getCurrent();
        Oea.invokeSvc({
            svc: this.svc,
            svcParams: {
                Model: c.get("EntityType"),
                ViewName: c.get("ViewName")
            },
            callback: function (res) {
                if (res.Success) {
                    //该方法为 CustomizeUI command 添加
                    view.reloadViewConfiguration();

                    //                    //动态变化还原最新配置的可见性
                    //                    var cmdCtl = view.getCmdControl('RestoreViewConfig');
                    //                    if (cmdCtl.hidden) { cmdCtl.show(); }
                    //                    else { cmdCtl.hide(); }
                }
            }
        });
    }
});

//command end

Ext.define('RestoreViewConfig', {
    extend: 'BackupViewConfig',
    config: {
        meta: {
            text: "还原",
            tooltip: "还原为最新的配置"
            //            ,hidden: true
        }
    },
    constructor: function () {
        this.svc = 'RestoreViewConfigService';
    }
});

//command end

Ext.define('OpenConfigFile', {
    extend: 'Oea.cmd.Command',
    config: {
        meta: {
            text: "XML",
            tooltip: "打开XML文件进行编辑",
            hidden: window.location.href.indexOf('localhost') < 0
        }
    },
    execute: function (view) {
        var c = view.getCurrent();
        Oea.invokeSvc({
            svc: 'GetBlockConfigFileService',
            svcParams: {
                Model: c.get("EntityType"),
                ViewName: c.get("ViewName")
            },
            callback: function (res) {
                if (!res.Opened) {
                    Ext.Msg.alert("提示", "暂时还没有进行任何配置，没有找到对应的 XML 文件。");
                }
            }
        });
    }
});

File: OEA/WPF/OEA.Module.WPF/Command/SysCommand/CustomizeUI/BackupViewConfig.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using OEA.MetaModel.Attributes;
using OEA.WPF.Command;
using SimpleCsla;
using OEA.Library.Modeling.Web;
using OEA.Library;
using OEA.Module.WPF;

namespace OEA.WPF.Command
{
    [Command(Label = "重置", ToolTip = "重置为原始配置")]
    public class BackupViewConfig : ViewCommand
    {
        protected virtual BackupViewConfigService CreateSVC()
        {
            return new BackupViewConfigService();
        }

        public override void Execute(ObjectView view)
        {
            var c = view.Current as ViewConfigurationModel;
            var svc = this.CreateSVC();
            svc.Model = c.EntityType;
            svc.ViewName = c.ViewName;
            svc.Invoke();

            view.Current = RF.Concreate<ViewConfigurationModelRepository>()
                .GetByName(new ViewConfigurationModelNameCriteria
                {
                    EntityType = c.EntityType,
                    ViewName = c.ViewName
                });
        }
    }

    [Command(Label = "还原", ToolTip = "还原为最新的配置")]
    public class RestoreViewConfig : BackupViewConfig
    {
        protected override BackupViewConfigService CreateSVC()
        {
            return new RestoreViewConfigService();
        }
    }

    [Command(Label = "XML", ToolTip = "打开XML文件进行编辑")]
    public class OpenConfigFile : ViewCommand
    {
        public override void Execute(ObjectView view)
        {
            var c = view.Current as ViewConfigurationModel;
            var svc = new GetBlockConfigFileService();
            svc.Model = c.EntityType;
            svc.ViewName = c.ViewName;
            svc.Invoke(out svc);

            if (!svc.Opened)
            {
                App.Current.MessageBox.Show("提示", "暂时还没有进行任何配置，没有找到对应的 XML 文件。");
            }
        }
    }
}